contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -1185,6 +1191,11 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
     )
         internal
     {
+        // We don't set storage if the value didn't change.
+        if (_getContractStorage(_contract, _key) == _value) {
+            return;
+        }
+
```

Comment:
Is this an optimization or was it somehow a requirement?  If requirement, can you specify why in the comment?  Not obvious to me why this is being done

Reply:
Ah sorry yes -- this is a requirement. Reason is that if we change the storage slot, then we mark it as "changed" in the state manager and it has to be committed into the state trie. Normally this isn't a problem (because updating a value to the same value won't change the root) *except* in the case that the state trie was originally empty and a contract attempts to store `0x00.....`. This causes a discrepancy because the default storage root is `keccak256('80')` but once you store a value (even if it's `0x00...`), that storage root changes. Will reflect in a comment.